### ISSUE TITLE
OSError 104 ECONNRESET: Authentication failed

### ISSUE THREAD

**Issue by martinorob** [bimmerconnected/bimmer_connected#208]
I'm not be able to use the api.. 
even if I change the password.. 
(I'm able to login on bmw connect drive portal and on my iOS app)

`./bimmerconnected vehiclefinder myemail mypassword rest_of_world myvin
DEBUG:bimmer_connected.account:Getting vehicle list
DEBUG:bimmer_connected.account:getting new oauth token
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): customer.bmwgroup.com
ERROR:bimmer_connected.account:Authentication failed. Maybe your password is invalid?
ERROR:bimmer_connected.account:('Connection aborted.', OSError("(104, 'ECONNRESET')"))
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 598, in urlopen
    httplib_response = self._make_request(conn, method, url,
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 387, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 383, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.8/http/client.py", line 1347, in getresponse
    response.begin()
  File "/usr/lib/python3.8/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.8/http/client.py", line 268, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/contrib/pyopenssl.py", line 285, in recv_into
    raise SocketError(str(e))
OSError: (104, 'ECONNRESET')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 430, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 638, in urlopen
    retries = retries.increment(method, url, error=e, _pool=self,
  File "/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py", line 357, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/packages/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 598, in urlopen
    httplib_response = self._make_request(conn, method, url,
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 387, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 383, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.8/http/client.py", line 1347, in getresponse
    response.begin()
  File "/usr/lib/python3.8/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.8/http/client.py", line 268, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/contrib/pyopenssl.py", line 285, in recv_into
    raise SocketError(str(e))
urllib3.exceptions.ProtocolError: ('Connection aborted.', OSError("(104, 'ECONNRESET')"))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/myuser/.local/lib/python3.8/site-packages/bimmer_connected/account.py", line 111, in _get_oauth_token
    response = self.send_request(url, data=data, headers=headers, allow_redirects=False,
  File "/home/myuser/.local/lib/python3.8/site-packages/bimmer_connected/account.py", line 154, in send_request
    response = requests.post(url, headers=headers, data=data, allow_redirects=allow_redirects,
  File "/usr/local/lib/python3.8/dist-packages/requests/api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 490, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', OSError("(104, 'ECONNRESET')"))
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 598, in urlopen
    httplib_response = self._make_request(conn, method, url,
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 387, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 383, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.8/http/client.py", line 1347, in getresponse
    response.begin()
  File "/usr/lib/python3.8/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.8/http/client.py", line 268, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/contrib/pyopenssl.py", line 285, in recv_into
    raise SocketError(str(e))
OSError: (104, 'ECONNRESET')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 430, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 638, in urlopen
    retries = retries.increment(method, url, error=e, _pool=self,
  File "/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py", line 357, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/packages/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 598, in urlopen
    httplib_response = self._make_request(conn, method, url,
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 387, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 383, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/lib/python3.8/http/client.py", line 1347, in getresponse
    response.begin()
  File "/usr/lib/python3.8/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.8/http/client.py", line 268, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/contrib/pyopenssl.py", line 285, in recv_into
    raise SocketError(str(e))
urllib3.exceptions.ProtocolError: ('Connection aborted.', OSError("(104, 'ECONNRESET')"))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/myuser/.local/lib/python3.8/site-packages/bimmer_connected/account.py", line 111, in _get_oauth_token
    response = self.send_request(url, data=data, headers=headers, allow_redirects=False,
  File "/home/myuser/.local/lib/python3.8/site-packages/bimmer_connected/account.py", line 154, in send_request
    response = requests.post(url, headers=headers, data=data, allow_redirects=allow_redirects,
  File "/usr/local/lib/python3.8/dist-packages/requests/api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 490, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', OSError("(104, 'ECONNRESET')"))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./bimmerconnected", line 8, in <module>
    sys.exit(main())
  File "/home/myuser/.local/lib/python3.8/site-packages/bimmer_connected/cli.py", line 240, in main
    args.func(args)
  File "/home/myuser/.local/lib/python3.8/site-packages/bimmer_connected/cli.py", line 140, in vehicle_finder
    account = ConnectedDriveAccount(args.username, args.password, get_region_from_name(args.region))
  File "/home/myuser/.local/lib/python3.8/site-packages/bimmer_connected/account.py", line 61, in __init__
    self._get_vehicles()
  File "/home/myuser/.local/lib/python3.8/site-packages/bimmer_connected/account.py", line 226, in _get_vehicles
    self._get_oauth_token()
  File "/home/myuser/.local/lib/python3.8/site-packages/bimmer_connected/account.py", line 117, in _get_oauth_token
    raise OSError(msg) from exception
OSError: Authentication failed. Maybe your password is invalid?
`

**Comment by janhaa** [comment]
No, I have the same error. It seems like BMW messed something up with their certificates. I guess/hope they will fix it soon.

```$ curl https://customer.bmwgroup.com
curl: (51) SSL: certificate subject name (grpw-prod-a1.bmw.com) does not match target host name 'customer.bmwgroup.com'
```

**Comment by martinorob** [comment]
@janhaa thank you! 
Keep in touch please

**Comment by martinorob** [comment]
may be some "ignore certificate" that we can add to che script?

**Comment by janhaa** [comment]
@martinorob not a viable solution in my opinion, as it allows for MITM.

**Comment by martinorob** [comment]
of course.. it was only as workaround for this moment.. 😊

**Comment by janhaa** [comment]
It seems they like changed IP addresses. From where I try to access the BMW API it is "160.46.231.195" but on my local machine it is "35.246.215.17".
So clearing DNS cache should be the solution.

**Comment by martinorob** [comment]
I just resolve it with 35.246.215.17 and I have the same error...

**Comment by martinorob** [comment]
https://downforeveryoneorjustme.com/customer.bmwgroup.com

DOWN

**Comment by janhaa** [comment]
Well, customer.bmwgroup.com does not return anything on its default / route. Therefore its probably marked as down by that site. I was able to resolve the connection issue by my comment above.

**Comment by martinorob** [comment]
so you have solved using 35.246.215.17  as ip for customer.bmwgroup.com right?

**Comment by janhaa** [comment]
Yes. I am manually using "https://35.246.215.17/gcdm/oauth/authenticate" with Host Header "customer.bmwgroup.com" and ignoring certificate errors in my code (I am not using the code in this repository).

**Comment by martinorob** [comment]
Do you use rest_of_world?

**Comment by janhaa** [comment]
I am not using the code from this repository, I was only inspired by it, thats why I was checking here for someone else having the issue :) Sorry can't help you there but the above server should work for getting access token for APIs.

**Comment by LukasSzmidt** [comment]
Hey guys, I have the same issue as OP. Using repo code, nothing custom.

bmw_connected_drive:
  G20: 
    username: !secret bmw_connected_drive_username
    password: !secret bmw_connected_drive_password
    region: rest_of_world

My DNS is:
Address:  35.246.215.17
Name:    rgeu3rt001-1-routers.dn.apigee.net
Aliases:  customer.bmwgroup.com
          bmw-rampup2-prod.apigee.net
          bmw-rampup2-dev.dn.apigee.net

Same issue as OP. Cleared DNS cache, rebooted HA host, no dice.

**Comment by martinorob** [comment]
https://www.sslshopper.com/ssl-checker.html#hostname=customer.bmwgroup.com

The SSL chain Is ok

and the SSL report has an A+ 
https://www.ssllabs.com/ssltest/analyze.html?d=customer.bmwgroup.com


host customer.bmwgroup.com
customer.bmwgroup.com is an alias for bmw-rampup2-prod.apigee.net.
bmw-rampup2-prod.apigee.net is an alias for bmw-rampup2-dev.dn.apigee.net.
bmw-rampup2-dev.dn.apigee.net is an alias for rgeu3rt001-1-routers.dn.apigee.net.
rgeu3rt001-1-routers.dn.apigee.net has address 35.246.215.17

**Comment by LukasSzmidt** [comment]
I don't think it's a DNS issue, to be honest. DNS could be location-dependant, so CNAME -> A would resolve to different IP's from different locations. First web based nslookup I found on Google returns the following 

Non-authoritative answer:
customer.bmwgroup.com	canonical name = bmw-rampup2-prod.apigee.net.
bmw-rampup2-prod.apigee.net	canonical name = bmw-rampup2-dev.dn.apigee.net.
bmw-rampup2-dev.dn.apigee.net	canonical name = agea1rt001-1-routers.dn.apigee.net.
Name:	agea1rt001-1-routers.dn.apigee.net
Address: 35.196.23.22

Cloudflare and Google DNS returns different IP's as well

C:\>nslookup customer.bmwgroup.com 1.1.1.1
Server:  one.one.one.one
Address:  1.1.1.1

Non-authoritative answer:
Name:    rgeu3rt001-1-routers.dn.apigee.net
Address:  35.246.215.17
Aliases:  customer.bmwgroup.com
          bmw-rampup2-prod.apigee.net
          bmw-rampup2-dev.dn.apigee.net


C:\>nslookup customer.bmwgroup.com 8.8.8.8
Server:  dns.google
Address:  8.8.8.8

Non-authoritative answer:
Name:    ageu1rt001-1-routers.dn.apigee.net
Address:  35.240.103.94
Aliases:  customer.bmwgroup.com
          bmw-rampup2-prod.apigee.net
          bmw-rampup2-dev.dn.apigee.net

**Comment by LukasSzmidt** [comment]
> https://www.sslshopper.com/ssl-checker.html#hostname=customer.bmwgroup.com
> 
> The SSL chain Is ok

I agree on this. My client returns correct chain and correect SAN's as well.

**Comment by GitHubxtsnl** [comment]
I'm in the Netherlands and I have the same issue since today.
I only restarted my HA, not the HOST, and this came up.

-----------
Log Details (ERROR)
Logger: homeassistant.setup
Source: components/bmw_connected_drive/__init__.py:127
First occurred: 3:40:35 PM (1 occurrences)
Last logged: 3:40:35 PM

Error during setup of component bmw_connected_drive
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 670, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 426, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 421, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/lib/python3.8/http/client.py", line 1347, in getresponse
    response.begin()
  File "/usr/local/lib/python3.8/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.8/http/client.py", line 276, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 726, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.8/site-packages/urllib3/util/retry.py", line 403, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/local/lib/python3.8/site-packages/urllib3/packages/six.py", line 734, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 670, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 426, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 421, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/lib/python3.8/http/client.py", line 1347, in getresponse
    response.begin()
  File "/usr/local/lib/python3.8/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.8/http/client.py", line 276, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 111, in _get_oauth_token
    response = self.send_request(url, data=data, headers=headers, allow_redirects=False,
  File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 154, in send_request
    response = requests.post(url, headers=headers, data=data, allow_redirects=allow_redirects,
  File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 213, in _async_setup_component
    result = await task
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/__init__.py", line 52, in setup
    accounts.append(setup_account(account_config, hass, name))
  File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/__init__.py", line 80, in setup_account
    cd_account = BMWConnectedDriveAccount(username, password, region, name, read_only)
  File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/__init__.py", line 127, in __init__
    self.account = ConnectedDriveAccount(username, password, region)
  File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 61, in __init__
    self._get_vehicles()
  File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 226, in _get_vehicles
    self._get_oauth_token()
  File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 117, in _get_oauth_token
    raise OSError(msg) from exception
OSError: Authentication failed. Maybe your password is invalid?
-----------

**Comment by epl2005** [comment]
I have exactly the same issue (rest_of_world region) .. any update on it ?

**Comment by martinorob** [comment]
No news.. we are waiting for @gerard33 😊

**Comment by janhaa** [comment]
Sorry for the confusion, was a different issue on my end then :)

**Comment by danfulton72** [comment]
Same issue in UK,

Using bimmer_connected on the command line as my 0.116.4 home assistant install reported an error on the bmw_connected_drive integration, this is the output I see.

I know my password is correct, I've changed it to try and 'reset' the app and website both work for me, so I think maybe BMW have changed something (I'm in the UK).

DEBUG:bimmer_connected.account:Getting vehicle list
DEBUG:bimmer_connected.account:getting new oauth token
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): customer.bmwgroup.com:443
ERROR:bimmer_connected.account:Authentication failed. Maybe your password is invalid?
ERROR:bimmer_connected.account:('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
Traceback (most recent call last):
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 677, in urlopen
chunked=chunked,
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 426, in _make_request
six.raise_from(e, None)
File "", line 3, in raise_from
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 421, in _make_request
httplib_response = conn.getresponse()
File "/usr/lib/python3.6/http/client.py", line 1331, in getresponse
response.begin()
File "/usr/lib/python3.6/http/client.py", line 297, in begin
version, status, reason = self._read_status()
File "/usr/lib/python3.6/http/client.py", line 266, in _read_status
raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/home/dan/.local/lib/python3.6/site-packages/requests/adapters.py", line 449, in send
timeout=timeout
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 727, in urlopen
method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/util/retry.py", line 410, in increment
raise six.reraise(type(error), error, _stacktrace)
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/packages/six.py", line 734, in reraise
raise value.with_traceback(tb)
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 677, in urlopen
chunked=chunked,
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 426, in _make_request
six.raise_from(e, None)
File "", line 3, in raise_from
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 421, in _make_request
httplib_response = conn.getresponse()
File "/usr/lib/python3.6/http/client.py", line 1331, in getresponse
response.begin()
File "/usr/lib/python3.6/http/client.py", line 297, in begin
version, status, reason = self._read_status()
File "/usr/lib/python3.6/http/client.py", line 266, in _read_status
raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/home/dan/.local/lib/python3.6/site-packages/bimmer_connected/account.py", line 112, in _get_oauth_token
expected_response=expected_response_code, post=True)
File "/home/dan/.local/lib/python3.6/site-packages/bimmer_connected/account.py", line 155, in send_request
params=params)
File "/home/dan/.local/lib/python3.6/site-packages/requests/api.py", line 119, in post
return request('post', url, data=data, json=json, **kwargs)
File "/home/dan/.local/lib/python3.6/site-packages/requests/api.py", line 61, in request
return session.request(method=method, url=url, **kwargs)
File "/home/dan/.local/lib/python3.6/site-packages/requests/sessions.py", line 530, in request
resp = self.send(prep, **send_kwargs)
File "/home/dan/.local/lib/python3.6/site-packages/requests/sessions.py", line 643, in send
r = adapter.send(request, **kwargs)
File "/home/dan/.local/lib/python3.6/site-packages/requests/adapters.py", line 498, in send
raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
Traceback (most recent call last):
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 677, in urlopen
chunked=chunked,
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 426, in _make_request
six.raise_from(e, None)
File "", line 3, in raise_from
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 421, in _make_request
httplib_response = conn.getresponse()
File "/usr/lib/python3.6/http/client.py", line 1331, in getresponse
response.begin()
File "/usr/lib/python3.6/http/client.py", line 297, in begin
version, status, reason = self._read_status()
File "/usr/lib/python3.6/http/client.py", line 266, in _read_status
raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/home/dan/.local/lib/python3.6/site-packages/requests/adapters.py", line 449, in send
timeout=timeout
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 727, in urlopen
method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/util/retry.py", line 410, in increment
raise six.reraise(type(error), error, _stacktrace)
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/packages/six.py", line 734, in reraise
raise value.with_traceback(tb)
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 677, in urlopen
chunked=chunked,
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 426, in _make_request
six.raise_from(e, None)
File "", line 3, in raise_from
File "/home/dan/.local/lib/python3.6/site-packages/urllib3/connectionpool.py", line 421, in _make_request
httplib_response = conn.getresponse()
File "/usr/lib/python3.6/http/client.py", line 1331, in getresponse
response.begin()
File "/usr/lib/python3.6/http/client.py", line 297, in begin
version, status, reason = self._read_status()
File "/usr/lib/python3.6/http/client.py", line 266, in _read_status
raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/home/dan/.local/lib/python3.6/site-packages/bimmer_connected/account.py", line 112, in _get_oauth_token
expected_response=expected_response_code, post=True)
File "/home/dan/.local/lib/python3.6/site-packages/bimmer_connected/account.py", line 155, in send_request
params=params)
File "/home/dan/.local/lib/python3.6/site-packages/requests/api.py", line 119, in post
return request('post', url, data=data, json=json, **kwargs)
File "/home/dan/.local/lib/python3.6/site-packages/requests/api.py", line 61, in request
return session.request(method=method, url=url, **kwargs)
File "/home/dan/.local/lib/python3.6/site-packages/requests/sessions.py", line 530, in request
resp = self.send(prep, **send_kwargs)
File "/home/dan/.local/lib/python3.6/site-packages/requests/sessions.py", line 643, in send
r = adapter.send(request, **kwargs)
File "/home/dan/.local/lib/python3.6/site-packages/requests/adapters.py", line 498, in send
raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "/home/dan/.local/bin/bimmerconnected", line 11, in
sys.exit(main())
File "/home/dan/.local/lib/python3.6/site-packages/bimmer_connected/cli.py", line 240, in main
args.func(args)
File "/home/dan/.local/lib/python3.6/site-packages/bimmer_connected/cli.py", line 116, in fingerprint
log_responses=time_dir)
File "/home/dan/.local/lib/python3.6/site-packages/bimmer_connected/account.py", line 61, in init
self._get_vehicles()
File "/home/dan/.local/lib/python3.6/site-packages/bimmer_connected/account.py", line 226, in _get_vehicles
self._get_oauth_token()
File "/home/dan/.local/lib/python3.6/site-packages/bimmer_connected/account.py", line 117, in _get_oauth_token
raise OSError(msg) from exception
OSError: Authentication failed. Maybe your password is invalid?

**Comment by rikroe** [comment]
Either BMW is experiencing some general API issues (but then the official apps probably wouldn't work either) or has changed something in the way we login to the acount. It could be a change to the process or just to some of the pre-baked credentials that are supplied within the app.

I'll try to have a look.

**Comment by rikroe** [comment]
Fun fact: If one is using the "old" API to get the token (`bimmer_connected<0.7.0`) it still works. However the new features in the master branch (weeklyPlanner etc.) won't work. 
If you want to do a quick fix on your side, you can edit `account.py` as follows: https://github.com/bimmerconnected/bimmer_connected/compare/master...rikroe:fix/connection_refused
**This might break some other stuff!!!**

Maybe it is only an API error and everything is working again tomorrow. I'll let you know if I find out more!

**Comment by martinorob** [comment]
> Fun fact: If one is using the "old" API to get the token (`bimmer_connected<0.7.0`) it still works. However the new features in the master branch (weeklyPlanner etc.) won't work.
> If you want to do a quick fix on your side, you can edit `account.py` as follows: [master...rikroe:fix/connection_refused](https://github.com/bimmerconnected/bimmer_connected/compare/master...rikroe:fix/connection_refused)
> **This might break some other stuff!!!**
> 
> Maybe it is only an API error and everything is working again tomorrow. I'll let you know if I find out more!

Thank you!!!!
It's works!!!! 

NB: is a workaround but it works!

**Comment by scstraus** [comment]
Same issue in Czechia..  I guess I'll give it a while to see if a fix is found before going workaround.

**Comment by danfulton72** [comment]
> Fun fact: If one is using the "old" API to get the token (`bimmer_connected<0.7.0`) it still works. However the new features in the master branch (weeklyPlanner etc.) won't work.
> If you want to do a quick fix on your side, you can edit `account.py` as follows: [master...rikroe:fix/connection_refused](https://github.com/bimmerconnected/bimmer_connected/compare/master...rikroe:fix/connection_refused)
> **This might break some other stuff!!!**
> 
> Maybe it is only an API error and everything is working again tomorrow. I'll let you know if I find out more!

Workaround worked for me, thanks

**Comment by epl2005** [comment]
> Fun fact: If one is using the "old" API to get the token (`bimmer_connected<0.7.0`) it still works. However the new features in the master branch (weeklyPlanner etc.) won't work.
> If you want to do a quick fix on your side, you can edit `account.py` as follows: [master...rikroe:fix/connection_refused](https://github.com/bimmerconnected/bimmer_connected/compare/master...rikroe:fix/connection_refused)
> **This might break some other stuff!!!**
> 
> Maybe it is only an API error and everything is working again tomorrow. I'll let you know if I find out more!

As this morning nothing changed, I applied the workaround and also in my case it works !
I noticed accessing by the app that password policy changed and also a pin has been required, I don't think this explain the issue but it is the evidence  they changed something at their side.
Thank You

**Comment by martinorob** [comment]
> Excuse me, could you tell me where the file is located? I look in the HA folder and I can't find it
> thx!

On my server
/home/username/.local/lib/python3.8/site-packages/bimmer_connected

**Comment by danibiza** [comment]
> > Excuse me, could you tell me where the file is located? I look in the HA folder and I can't find it
> > thx!
> 
> On my server
> /home/username/.local/lib/python3.8/site-packages/bimmer_connected

forgive my ignorance, but is that under Docker? I am with HASSIO and I cannot find a way to enter that route. Is it possible or do I have to resign? Thanks friend

**Comment by danfulton72** [comment]
> forgive my ignorance, but is that under Docker? I am with HASSIO and I cannot find a way to enter that route. Is it possible or do I have to resign? Thanks friend

I'm using core, so don't know about HASSIO, but this is how I edited the file, 

Get a command line in the container via docker exec (here home-assistant is the name of my container)

docker exec -it home-assistant bash

Then in the container the file is located in this directory

/usr/local/lib/python3.8/site-packages/bimmer_connected

**Comment by danibiza** [comment]
> > forgive my ignorance, but is that under Docker? I am with HASSIO and I cannot find a way to enter that route. Is it possible or do I have to resign? Thanks friend
> 
> I'm using core, so don't know about HASSIO, but this is how I edited the file,
> 
> Get a command line in the container via docker exec (here home-assistant is the name of my container)
> 
> docker exec -it home-assistant bash
> 
> Then in the container the file is located in this directory
> 
> /usr/local/lib/python3.8/site-packages/bimmer_connected

I have tried to enter through ssh and I see some more folders and using search commands etc, I cannot find that file or folder. I don't know if it can't or am I doing something wrong. This image is my HA directory.

![image](https://user-images.githubusercontent.com/6940922/96716993-1c95a480-13a6-11eb-8945-8d01c51afdc8.png)

**Comment by deluxestyle** [comment]
thank you. Workaround is working for me

**Comment by jsimon3** [comment]
@danibiza , you have do do as @danfulton72 suggested: If you ssh in, you land in the ssh container. From there you need to get into the container thats running HA: 
> Get a command line in the container via docker exec (here home-assistant is the name of my container)
>
> docker exec -it home-assistant bash
>
> Then in the container
>
> the file is located in this directory
>
> /usr/local/lib/python3.8/site-packages/bimmer_connected

If you are not sure about the name of the container, you can list them all with
_docker ps -a_

**Comment by luka6000** [comment]
I can confirm posted workaround is working. Thank you!

**Comment by rikroe** [comment]
So BMW did really change stuff in how the API is expecting stuff... Good thing I have my old, trusted and rooted HTC One lying around so I was able to convince the current BMW app to allow some man-in-the-middle.

#210 contains a quick fix that should enable everything to work again. 

Please test, if it works for you I'll create a bugfix release for PyPI.

**Comment by danfulton72** [comment]
> So BMW did really change stuff in how the API is working... Good thing I have my old, trusted and rooted HTC One lying around so I was able to convince the current BMW app to allow some man-in-the-middle sniffing.
> 
> #210 contains a quick fix that should enable everything to work again.
> 
> Please test, if it works for you I'll create a bugfix release for PyPI.

I've applied the changes to account.py and const.py and the integration is working, so looks good.

**Comment by danibiza** [comment]
Under Hassio and ssh I can't find a way to get to that file to modify it, I even do searches by file or folder name with whereis or which and it doesn't find anything on the entire server.

**Comment by danfulton72** [comment]
@danibiza looks like you'll need to wait for the change above to work it's way through the system.

**Comment by deluxestyle** [comment]
> So BMW did really change stuff in how the API is working... Good thing I have my old, trusted and rooted HTC One lying around so I was able to convince the current BMW app to allow some man-in-the-middle sniffing.
> 
> #210 contains a quick fix that should enable everything to work again.
> 
> Please test, if it works for you I'll create a bugfix release for PyPI.

I applied the change as well and it is working fine

**Comment by deluxestyle** [comment]
> Under Hassio and ssh I can't find a way to get to that file to modify it, I even do searches by file or folder name with whereis or which and it doesn't find anything on the entire server.

are you sure that you are in the docker image?

**Comment by rikroe** [comment]
[bimmer_connected 0.7.8](https://github.com/bimmerconnected/bimmer_connected/releases/tag/0.7.8) has been published to PyPI. 
When using HA you should now be able to manually update by `pip3 install bimmer-connected==0.7.8`. I will create a PR for HA later today.

**Comment by danibiza** [comment]
> [bimmer_connected 0.7.8](https://github.com/bimmerconnected/bimmer_connected/releases/tag/0.7.8) has been published to PyPI.
> When using HA you should now be able to manually update by `pip3 install bimmer-connected==0.7.8`. I will create a PR for HA later today.

I have updated with "pip3 install bimmer-connected == 0.7.8" it says that it was installed correctly, I have restarted HA and the raspberry but it still does not work. That could be happening?

**Comment by gerard33** [comment]
@danibiza You have to update the version in manifest.json of the BMW integration to `0.7.8` as well, otherwise HA will still use `0.7.7`.

**Comment by jsimon3** [comment]
Ha, I did not know that either. I was wondering why the update would work ok, but after the restart I'd be back on 0.7.7

Thanks for the tip, all good now.

**Comment by danibiza** [comment]
nothing there is no way, I have searched for manifest.json but it does not come out anywhere. An option that I have installed is to go back to 0.7.7 to modify the files account.py and const.py by hand with the changes indicated but it still does not work. I'm desperate, I don't know what else to do ...

**Comment by gerard33** [comment]
You can either wait for this to be merged in HA (PR is made) or copy the integration as a custom_component where you can change the manifest.json file.

**Comment by danibiza** [comment]
> You can either wait for this to be merged in HA (PR is made) or copy the integration as a custom_component where you can change the manifest.json file.

I have downloaded the package https://github.com/bimmerconnected/ha_custom_component and put it in my custom_components folder, I have modified the manifest.json with "bimmer_connected == 0.7.8" but it still does not work. I have seen that in the github instructions it says that you have to unzip the bmw_connected_drive folder in custom_components, but inside the zip it is named as bmw_connecteddrive (without the underscore) and inside the manifest.json the documentation path also comes out with bmw_connecteddrive. Is this so or is it a misprint?

thanks for your help gerard33

**Comment by jsimon3** [comment]
> nothing there is no way, I have searched for manifest.json but it does not come out anywhere. An option that I have installed is to go back to 0.7.7 to modify the files account.py and const.py by hand with the changes indicated but it still does not work. I'm desperate, I don't know what else to do ...

I would bet anything that you just ssh'd in and did not follow the instructions to get into the right container. You can search until you're blue in the face, but you will not find manifest.json or the bimmer_connected directory or anything.

Do a simple test, try _pip show bimmer_connected_
You should get:
_Version: 0.7.8
Summary: Library to read data from the BMW Connected Drive portal
Home-page: https://github.com/bimmerconnected/bimmer_connected
Author: gerard33
Author-email: bietenbak@yahoo.com
License: Apache-2
Location: /usr/local/lib/python3.8/site-packages
Requires: requests
Required-by:_ 

If you get  _WARNING: Package(s) not found: bimmer_connected_ then you are in the wrong container.

**Comment by rikroe** [comment]
The custom component is not up to date to the current version. It has additional features that are part of home-assistant/core#39585.
However I will update the custom component accordingly so you should be able to use that for now.

**Comment by rikroe** [comment]
Until the hotfix is in 0.117, you can use our custom component. Download at https://github.com/bimmerconnected/ha_custom_component/releases/tag/20201022.1 or via HACS ;)

__Please note that it includes a config flow for the component which is not part of the main HA branch. So if you go the custom component way you need to stay on it until home-assistant/core#39585 is merged!!__

**Comment by danibiza** [comment]
> > nothing there is no way, I have searched for manifest.json but it does not come out anywhere. An option that I have installed is to go back to 0.7.7 to modify the files account.py and const.py by hand with the changes indicated but it still does not work. I'm desperate, I don't know what else to do ...
> 
> I would bet anything that you just ssh'd in and did not follow the instructions to get into the right container. You can search until you're blue in the face, but you will not find manifest.json or the bimmer_connected directory or anything.
> 
> Do a simple test, try _pip show bimmer_connected_
> You should get:
> _Version: 0.7.8 Summary: Library to read data from the BMW Connected Drive portal Home-page: https://github.com/bimmerconnected/bimmer_connected Author: gerard33 Author-email: [bietenbak@yahoo.com](mailto:bietenbak@yahoo.com) License: Apache-2 Location: /usr/local/lib/python3.8/site-packages Requires: requests Required-by:_
> 
> If you get _WARNING: Package(s) not found: bimmer_connected_ then you are in the wrong container.



> Until the hotfix is in 0.117, you can use our custom component. Download at https://github.com/bimmerconnected/ha_custom_component/releases/tag/20201022.1 or via HACS ;)
> 
> **Please note that it includes a config flow for the component which is not part of the main HA branch. So if you go the custom component way you need to stay on it until [home-assistant/core#39585](https://github.com/home-assistant/core/pull/39585) is merged!!**

You were right, I was not where I had to be ... I'm very stubborn, sorry. Putting pip show bimmer_connected was giving the error "WARNING: Package (s) not found: bimmer_connected" after that I have put docker exec -it homeassistant bash and already here I have put pip show bimmer_connected again, here I have already got the information that you have happened to me but with Version: 0.7.7. So I have put pip3 install bimmer-connected == 0.7.8 it has installed it correctly and I have put back pip show bimmer_connected to test if I had changed the version and it has been. Now I have installed with HACS the component that you have given me https://github.com/bimmerconnected/ha_custom_component/releases/tag/20201022.1

Now everything seems to be working fine.
I have a doubt, when version 0.117 comes out, removing the HACS component is enough, right?

forgive my ignorance and I'm sorry I wasted so much time with me. Although now I have learned something else, thank you!

**Comment by rikroe** [comment]
No worries at all, happy to help 👍 

It could be that you will get some additional errors in 0.117 when removing the component from HACS as it does include a new config flow which changes some of the base configuration of the component. We're waiting for another PR to be merged until it is safe to go back (it should be included in 0.118 probably).

Nevertheless just try to remove it after 0.117 and if you see too many errors, just get the custom component back ;)

**Comment by danibiza** [comment]
> No se preocupe en absoluto, feliz de ayudar 👍
> 
> Es posible que obtenga algunos errores adicionales en 0.117 cuando elimine el componente de HACS, ya que incluye un nuevo flujo de configuración que cambia parte de la configuración básica del componente. Estamos esperando que se fusione otro RP hasta que sea seguro volver atrás (probablemente debería incluirse en 0.118).
> 
> Sin embargo, intente eliminarlo después de 0.117 y si ve demasiados errores, simplemente recupere el componente personalizado;)

Thank you so much for everything!
I have noticed that everything works correctly except the location of the vehicle that continuously says "Awaiting Update". The fuel data, doors etc etc if updated, but the geolocation data does not. That could be happening? thx

**Comment by gerard33** [comment]
You are using the custom component?
Then please check if there is a second device_tracker for your car ending with _2. That is the one that will be updated.
If that's the case you can remove the old one and rename the new one.

**Comment by GitHubxtsnl** [comment]
> I'm in the Netherlands and I have the same issue since today.
> I only restarted my HA, not the HOST, and this came up.
> 
> Log Details (ERROR)
> Logger: homeassistant.setup
> Source: components/bmw_connected_drive/**init**.py:127
> First occurred: 3:40:35 PM (1 occurrences)
> Last logged: 3:40:35 PM
> 
> Error during setup of component bmw_connected_drive
> Traceback (most recent call last):
> File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 670, in urlopen
> httplib_response = self._make_request(
> File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 426, in _make_request
> six.raise_from(e, None)
> File "", line 3, in raise_from
> File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 421, in _make_request
> httplib_response = conn.getresponse()
> File "/usr/local/lib/python3.8/http/client.py", line 1347, in getresponse
> response.begin()
> File "/usr/local/lib/python3.8/http/client.py", line 307, in begin
> version, status, reason = self._read_status()
> File "/usr/local/lib/python3.8/http/client.py", line 276, in _read_status
> raise RemoteDisconnected("Remote end closed connection without"
> http.client.RemoteDisconnected: Remote end closed connection without response
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
> File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 439, in send
> resp = conn.urlopen(
> File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 726, in urlopen
> retries = retries.increment(
> File "/usr/local/lib/python3.8/site-packages/urllib3/util/retry.py", line 403, in increment
> raise six.reraise(type(error), error, _stacktrace)
> File "/usr/local/lib/python3.8/site-packages/urllib3/packages/six.py", line 734, in reraise
> raise value.with_traceback(tb)
> File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 670, in urlopen
> httplib_response = self._make_request(
> File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 426, in _make_request
> six.raise_from(e, None)
> File "", line 3, in raise_from
> File "/usr/local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 421, in _make_request
> httplib_response = conn.getresponse()
> File "/usr/local/lib/python3.8/http/client.py", line 1347, in getresponse
> response.begin()
> File "/usr/local/lib/python3.8/http/client.py", line 307, in begin
> version, status, reason = self._read_status()
> File "/usr/local/lib/python3.8/http/client.py", line 276, in _read_status
> raise RemoteDisconnected("Remote end closed connection without"
> urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
> File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 111, in _get_oauth_token
> response = self.send_request(url, data=data, headers=headers, allow_redirects=False,
> File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 154, in send_request
> response = requests.post(url, headers=headers, data=data, allow_redirects=allow_redirects,
> File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 119, in post
> return request('post', url, data=data, json=json, **kwargs)
> File "/usr/local/lib/python3.8/site-packages/requests/api.py", line 61, in request
> return session.request(method=method, url=url, **kwargs)
> File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 530, in request
> resp = self.send(prep, **send_kwargs)
> File "/usr/local/lib/python3.8/site-packages/requests/sessions.py", line 643, in send
> r = adapter.send(request, **kwargs)
> File "/usr/local/lib/python3.8/site-packages/requests/adapters.py", line 498, in send
> raise ConnectionError(err, request=request)
> requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
> 
> The above exception was the direct cause of the following exception:
> 
> ## Traceback (most recent call last):
> File "/usr/src/homeassistant/homeassistant/setup.py", line 213, in _async_setup_component
> result = await task
> File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
> result = self.fn(*self.args, **self.kwargs)
> File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/**init**.py", line 52, in setup
> accounts.append(setup_account(account_config, hass, name))
> File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/**init**.py", line 80, in setup_account
> cd_account = BMWConnectedDriveAccount(username, password, region, name, read_only)
> File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/**init**.py", line 127, in **init**
> self.account = ConnectedDriveAccount(username, password, region)
> File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 61, in **init**
> self._get_vehicles()
> File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 226, in _get_vehicles
> self._get_oauth_token()
> File "/usr/local/lib/python3.8/site-packages/bimmer_connected/account.py", line 117, in _get_oauth_token
> raise OSError(msg) from exception
> OSError: Authentication failed. Maybe your password is invalid?

since I upgrade to 0.117.1 the BMW ConnectedDrive integration worked again.

**Comment by nichwang88** [comment]
0.7.8 did not fix login issue for "china".
HA 0.117.1 did not fix.
